pkg/api/auth.go

```diff
@@ -20,46 +25,89 @@ func authFail(w http.ResponseWriter, realm string, delay int) {
 	WriteJSON(w, http.StatusUnauthorized, NewError(UNAUTHORIZED))
 }
 
+// nolint (gocyclo) - we use closure making this a complex subroutine
 func BasicAuthHandler(c *Controller) mux.MiddlewareFunc {
 	realm := c.Config.HTTP.Realm
 	if realm == "" {
 		realm = "Authorization Required"
 	}
 	realm = "Basic realm=" + strconv.Quote(realm)
-	delay := c.Config.HTTP.Auth.FailDelay
 
-	if c.Config.HTTP.Auth.HTPasswd.Path == "" {
+	// no password based authN, if neither LDAP nor HTTP BASIC is enabled
+	if c.Config.HTTP.Auth == nil || (c.Config.HTTP.Auth.HTPasswd.Path == "" && c.Config.HTTP.Auth.LDAP == nil) {
 		return func(next http.Handler) http.Handler {
 			return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 				if c.Config.HTTP.AllowReadAccess &&
 					c.Config.HTTP.TLS.CACert != "" &&
 					r.TLS.VerifiedChains == nil &&
 					r.Method != "GET" && r.Method != "HEAD" {
-					authFail(w, realm, delay)
+					authFail(w, realm, 5)
 					return
 				}
-
 				// Process request
 				next.ServeHTTP(w, r)
 			})
 		}
 	}
 
 	credMap := make(map[string]string)
+	delay := c.Config.HTTP.Auth.FailDelay
```

Comment:
Ok but based on the above logic, note that c.Config.HTTP.Auth.FailDelay could still be nil?

Reply:
See comment above